### PR TITLE
Make the CreateSnapshot API actually take a snapshot

### PR DESCRIPTION
**Pull request by jrockway**
This moves the recovery package into the snapshot package, and changes the RPC to call CreateSnapshot instead of just creating an empty snapshot row.  The original package duplicated some stuff (the snapshot struct), so I've merged that into the one that already exists in the snapshot package.

I noticed more shared library dependency hell when running this "for real", so I've changed the container to be built with the full debian dependency set for postgresql-client-17.  This means that things like perl are in there now, because postgres' wrapper scripts are written in perl.  It's only about 30MB of data, though, so not a big deal.

To test recovery, this PR exposes `pachdev restore --id=<snapshot id>` to start a restore job.  It does not work yet.

Finally, I adjusted the pachctl commands to show more than 100 snapshots and to show the fileset containing the database dump, so it can be downloaded with storage.Fileset.ReadFileset.

### CODE REVIEW

File: src/internal/snapshot/recovery_test.go [hunk 1]
```diff
@@ -96,30 +98,35 @@ func TestCreateAndRestoreSnaphot(t *testing.T) {
...
 		t.Fatalf("WithTx: %v", err)
 	}
-	fsToken := fileset.Token(got.SQLDumpFileSetToken[:])
-	t.Logf("fileset handle: %v", fsToken.HexString())
-	got.SQLDumpFileSetToken = uuid.UUID{}
-	if diff := cmp.Diff(want, got); diff != "" {
+	t.Logf("fileset handle: %v", got.GetSqlDumpFilesetId())
+	var fsToken fileset.Token
+	if err := fsToken.Scan(got.GetSqlDumpFilesetId()); err != nil {
+		t.Fatalf("parse token %v: %v", got.GetSqlDumpFilesetId(), err)
```
[the remark this answers is not in the document]
Comment (jrockway) [reply src/internal/snapshot/recovery_test.go:125]: %v prints strings as strings and fmt.Stringers as fmt.Stringers.  There is rarely a need to use %s or %d in Go.

File: src/internal/snapshot/recovery.go [hunk 2]
```diff
@@ -93,13 +87,14 @@ func createSnapshotRow(ctx context.Context, tx *sqlx.Tx, s *fileset.Storage) (re
...
+	if err := tx.GetContext(ctx, &result, `insert into recovery.snapshots (chunkset_id, pachyderm_version, metadata) values ($1, $2, $3) returning id`, chunksetID, version.Version.Canonical(), metadata); err != nil {
 		return 0, "", errors.Wrap(err, "create snapshot row")
 	}
 	// Write out psql to create only this snapshot row.  (Pre-existing snapshot rows are dumped
 	// by pg_dump.)
-	b.WriteString("COPY recovery.snapshots (id, chunkset_id, pachyderm_version) FROM stdin;\n")
-	fmt.Fprintf(&b, "%v\t%v\t%v\n", int64(result), int64(chunksetID), version.Version.Canonical())
+	b.WriteString("COPY recovery.snapshots (id, chunkset_id, pachyderm_version, metadata) FROM stdin;\n")
+	js := `{}` // TODO(MLDM-142): escape JSON for this
+	fmt.Fprintf(&b, "%v\t%v\t%v\t%s\n", int64(result), int64(chunksetID), version.Version.Canonical(), js)
```
[the remark this answers is not in the document]
Comment (jrockway) [reply src/internal/snapshot/recovery.go:97]: Same idea as above, %v prints int64s as int64s.

File: src/internal/snapshot/recovery.go [hunk 3]
```diff
@@ -226,8 +214,13 @@ func (s *Snapshotter) dumpTx(ctx context.Context, tx *pachsql.Tx) (id SnapshotID
 	return id, fh, nil
 }
 
+// CreateSnapshotOptions controls snapshotting behavior.
+type CreateSnapshotOptions struct {
```
[the remark this answers is not in the document]
Comment (jrockway) [reply src/internal/snapshot/recovery.go:218]: There are likely other options that will be added in the future.  You can wrap an Options struct with a builder (`WithMetadata(x)` -> `Options{Metadata: X}`) but it's just makework code to type in that doesn't actually do anything.

File: src/internal/snapshot/recovery.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zhang-Muyang) [reply src/internal/snapshot/recovery.go:97]: Got it. There is some inconsistency. Sometimes it's used %d and sometimes it's %v.

**Review by Zhang-Muyang — APPROVED** [review]
The go code part looks good and ci passes. Since it's release blocking PR, I'll approve to unblock this. But if time allows, I still want to act as a rubber duck to go over some bazel and tar details with Jon later.

**Pull request merged**